File: packages/ts/tests/e2e/__fixtures__/interpret/dirsql.config.js
// Native .js config fixture for dirsql. Inside `packages/ts/` the nearest
// package.json declares `"type": "module"`, so .js here is ESM -- same
// `import` / `export default` syntax as .mjs.

import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { DirSQL } from "dirsql";

const here = dirname(fileURLToPath(import.meta.url));

export default new DirSQL({
  root: join(here, "data"),
  tables: [
    {
      ddl: "CREATE TABLE papers (title TEXT)",
      glob: "**/meta.json",
      extract: (path) => [JSON.parse(readFileSync(path, "utf8"))],
    },
  ],
});

File: packages/python/tests/e2e/__fixtures__/interpret/dirsql.config.py
"""Happy-path fixture for `dirsql interpret` integration tests.

Exposes a single `papers` table whose `extract` reads each `meta.json`
under the colocated `data/` tree and returns its parsed contents.
"""

import json
import os

from dirsql import DirSQL, Table


def _extract(path):
    with open(path, encoding="utf-8") as f:
        return [json.load(f)]


app = DirSQL(
    root=os.path.join(os.path.dirname(__file__), "data"),
    tables=[
        Table(
            ddl="CREATE TABLE papers (title TEXT)",
            glob="**/meta.json",
            extract=_extract,
        )
    ],
)
